src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -131,6 +170,13 @@ public static string GetFileName(string path)
             return path.Substring(offset, count);
         }
 
+        public static ReadOnlySpan<char> GetFileName(ReadOnlySpan<char> path)
```

Comment:
As before, do not duplicate the body  of other methods. Use AsReadOnlySpan.

Same in other cases. Zero duplication!

Reply:
This also has the huge benefit of making the existing tests go through your span based implementation, catching any typos.